### ISSUE TITLE
DeepLabV3+ model plateauing

### ISSUE THREAD

**Issue by iancullinane06** [iancullinane06/BTYSTE-2025#12]
DeepLabV3+ model is plateauing

**Comment by iancullinane06** [comment]
First attempt that led to plateau
BC - 4 LR - 1e-3 DeepLabV3+

![Image](https://github.com/user-attachments/assets/68c3be7d-03d4-4adb-84b3-5f12b3132a11)

I'm going to introduce a Learning Rate schedule, data normalization, a combined loss function, L2 regularization and early stopping.

**Comment by iancullinane06** [comment]
These new settings have prevented the plateauing but are still poorly tuned


![Image](https://github.com/user-attachments/assets/c0d22c23-5a62-4fa4-b376-302424519674)

**Comment by iancullinane06** [comment]
The new slightly tweaked settings have improved the model training.

![Image](https://github.com/user-attachments/assets/3cd87d6d-febd-4955-afd6-d379944c4ed2)

The blue line is the new settings, it has a higher learning rate at the beginning and a lower factor for the learning rate adjuster

**Comment by iancullinane06** [comment]
# Finally some success!

The is the morning after running the model for approximately 14 hours. It's incredible


The model early stopped as the patience was too low but this is what the training looked like


![Image](https://github.com/user-attachments/assets/b0e428a6-81c6-4e04-a933-acb90a58cca0)


Running this model to inference an image, I got this mask


![Image](https://github.com/user-attachments/assets/7f8718c6-68cb-4693-8aa3-8d3bf5695870)

**Comment by iancullinane06** [comment]
The results are replicable!

![Image](https://github.com/user-attachments/assets/8b2de39b-a159-44d1-b042-c36dca91f197)


![Image](https://github.com/user-attachments/assets/4c94af5d-b4b8-4b2b-ab67-04bd21f6871d)


The model has generalised a solution for recognising rhododendron but still struggles a little with images that don't contain any rhododendron

![Image](https://github.com/user-attachments/assets/40113eab-1474-4e95-af8a-f4930f37f0c5)
